src/backends/simple.ts

```diff
@@ -0,0 +1,442 @@
+import type {
+    DecodedFrames,
+    Packet,
+    Demuxer,
+    Decoder,
+    Filterer
+} from '@antoinemopa/beamcoder';
+import beamcoder from '@antoinemopa/beamcoder';
+import type { ImageData } from 'canvas';
+import { createImageData } from 'canvas';
+import { BaseExtractor } from '../BaseExtractor';
+import type { Extractor, ExtractorArgs, InterpolateMode } from '../../framefusion';
+import { DownloadVideoURL } from '../DownloadVideoURL';
+
+const LOG = false;
+
+const createDecoder = ({
+    demuxer,
+    streamIndex,
+    threadCount,
+}: {
+    demuxer: Demuxer;
+    streamIndex: number;
+    threadCount: number;
+}): Decoder => {
+    return beamcoder.decoder({
+        demuxer: demuxer,
+        width: demuxer.streams[streamIndex].codecpar.width,
+        height: demuxer.streams[streamIndex].codecpar.height,
+        stream_index: streamIndex,
+        pix_fmt: demuxer.streams[streamIndex].codecpar.format,
+        thread_count: threadCount,
+    });
+};
+
+const createFilter = async({
+    stream,
+    outputPixelFormat,
+    interpolateFps,
+    interpolateMode = 'fast',
+}: {
+    stream: beamcoder.Stream;
+    outputPixelFormat: string;
+    interpolateFps?: number;
+    interpolateMode?: InterpolateMode;
+}): Promise<beamcoder.Filterer> => {
+    if (!stream.codecpar.format) {
+        return null;
+    }
+
+    let filterSpec = [`[in0:v]format=${stream.codecpar.format}`];
+
+    if (interpolateFps) {
+        if (interpolateMode === 'high-quality') {
+            filterSpec = [...filterSpec, `minterpolate=fps=${interpolateFps}`];
+        }
+        else if (interpolateMode === 'fast') {
+            filterSpec = [...filterSpec, `fps=${interpolateFps}`];
+        }
+        else {
+            throw new Error(`Unexpected interpolation mode: ${interpolateMode}`);
+        }
+    }
+
+    filterSpec = [...filterSpec];
```

Comment:
```suggestion
```
It seems like this line does not do anything

Reply:
Yeah, I used to add stuff in there and forgot to delete the line 😅 We can remove indeed

Reply:
And I haven't touched this part yet. I just blindly copied it over. Removing this line.